### DISCUSSION TITLE
Create new node after releasing the edge drag

### DISCUSSION THREAD

**Discussion by kumarvikramaditya1** [xyflow/xyflow#1329] [Q&A]
Is there a way in which we can create a new node from a context menu which gets triggered on releasing an edge drag. Here is some more explanation.

1. There is a default node created.
2. Two create a new node:
    3. Drag an edge from the handle and release it in the open space.
    4. A context menu appears asking a type of node selection
    5. On selecting a type, the new node is created and the Source and Target nodes are auto linked.

Any help is appreciated. Thanks !

**Comment by moklick** [comment]
You could try to overlay the handle with a custom context menu trigger component and when the connection is added, you remove the context menu trigger.

**Comment by samuellawerentz** [comment]
Did we find a way for this? I want to show a menu list on connectEnd. Then the user can select an item from the menu and then a new node will be added to the canvas?

The connection line should not disappear until the user has selected an item from the menu or clicks outside the menu, in which case the menu would close and line would disappear.

I just want to know how to keep the connection line in place and show a menu on drag end ("onConnectionEnd")

How to achieve this?

**Reply by samuellawerentz** [reply]
@moklick tagging you for visiblity.

**Reply by samuellawerentz** [reply]
Need something similar to this example - [Add Node On Edge Drop](https://reactflow.dev/examples/nodes/add-node-on-edge-drop)

But the user would be allowed to select which node gets added by choosing it from a menu.

My take is, we should add a temp node, which will host the menubar, once an item is selected, update the node type so that the node is auto linked.

Pls let me know if there is an alternative solution. Thanks!
